File: 프로그래머스/lv1/72410. 신규 아이디 추천/신규 아이디 추천.java
class Solution {
    public String solution(String new_id) {
        new_id = new_id.toLowerCase()
            .replaceAll("[^a-z0-9-_.]", "").replaceAll("\\.+", "\\.").replaceAll("^\\.|\\.$", "");
        if(new_id.isEmpty()) new_id = new_id + "a";
        if(new_id.length() >= 16) {
        	new_id = new_id.substring(0,15);
        	if(new_id.charAt(14) == '.') {
        		new_id = new_id.substring(0,14);
        	}
        }
        while(new_id.length() <= 2) {
        	new_id = new_id + new_id.substring(new_id.length()-1, new_id.length());
        }
        return new_id;
    }
}

File: 프로그래머스/1/72410. 신규 아이디 추천/신규 아이디 추천.js
function solution(new_id) {
    var answer = '';
    new_id = new_id.toLowerCase();
    for (let i = 0; i < new_id.length; i++){
        let ch = new_id.charAt(i);
        if ((ch >= 'a' && ch <= 'z') || (ch >= '0' && ch <= '9') || ch === '-' || ch === '_' || ch === '.')
            answer += ch;
    }
    while (answer.indexOf("..") != -1) answer = answer.replace("..", ".");
    while (answer.charAt(0) === '.') answer = answer.substring(1);
    while (answer.charAt(answer.length - 1) === '.') answer = answer.substring(0, answer.length - 1);
    if (answer === '') answer += 'a';
    answer = answer.slice(0, 15);
    if (answer.charAt(answer.length - 1) === '.') answer = answer.substring(0, answer.length - 1);
    while (answer.length < 3) answer += answer.charAt(answer.length - 1);
    return answer;
}